build_tools/upload_package_gforge.py

```diff
@@ -0,0 +1,47 @@
+import os
+import sys
+import glob
+import paramiko
+from OpenSSL.crypto import load_privatekey, FILETYPE_PEM, dump_privatekey
+
+# PLEASE BE SURE THESE VARIABLES ARE CORRECTLY DEFINED
+# variables
+server = os.environ['deploy_host']
+username = os.environ['deploy_user']
+password = os.environ['deploy_password']
+remotepath = os.environ['deploy_folder']
+key_filename = os.path.join(os.environ['src_dir'],'build_tools','openmeeg_deploy_key.pem')
+
+if __name__ == '__main__':
+    if len(sys.argv)==1:
+        print('nothing to upload')
+        sys.exit(0)
+    # load the encrypted file
+    with open(key_filename,'rb') as f:
+        priv = f.read()
+
+    pkey = load_privatekey(FILETYPE_PEM, priv, passphrase=password)
+    priv = dump_privatekey(FILETYPE_PEM, pkey)
+
+    # hack on the header
+    priv = priv.replace('BEGIN','BEGIN RSA')
+
+    key_filename += '.dec'
+    f.close()
+    f = open(key_filename, 'w')
+    f.write(priv)
+    f.close()
+
+    ssh = paramiko.SSHClient()
+    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    ssh.connect(server, username=username, key_filename=key_filename)
+    sftp = ssh.open_sftp()
+
+    for i in sys.argv[1:]:
+        for fname in glob.glob(i):
+            print('uploading file: ' + fname + ' ...')
+            sftp.put(fname, os.path.join(remotepath, fname))
+
+    sftp.close()
```

Comment:
mmmm..... I don't want to change that... not seen elsewhere on http://docs.paramiko.org/en/2.1/api/agent.html .... I doubt it would work (__enter__() ??. ..etc)